### PR TITLE
test(simple-color-picker): convert cypress tests to playwright

### PR DESCRIPTION
**Pull request by robinzigmond**
### Proposed behaviour

SimpleColorPicker should have Playwright tests

### Current behaviour

SimpleColorPicker does not have Playwright tests

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by nineteen88 — APPROVED** [review]

File: src/components/simple-color-picker/simple-color-picker.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+          await expect(simpleColorPickerInput(page, 0)).toBeChecked();
+        } else {
+          await expect(simpleColorPickerInput(page, 0)).not.toBeChecked();
+        }
+      });
+    });
+  }
+);
+
+test.describe("Should render SimpleColor component and check events", () => {
```
Comment (ZhuoyuJin) [inline src/components/simple-color-picker/simple-color-picker.pw.tsx:611]:
```suggestion
test.describe("check events for SimpleColor component", () => {
```

File: src/components/simple-color-picker/simple-color-picker.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+      const colorInput = simpleColorPickerInput(page, 5);
+      await colorInput.focus();
+      await colorInput.blur();
+      await expect(callbackCount).toBe(0);
+    });
+  }
+);
+
+test.describe(
+  "Should render SimpleColor component and check functionality",
```
Comment (ZhuoyuJin) [inline src/components/simple-color-picker/simple-color-picker.pw.tsx:565]:
```suggestion
  "check functionality for SimpleColor component",
```

File: src/components/simple-color-picker/simple-color-picker.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+        await expect(
+          simpleColorDiv(page, 0).locator("..").locator("..")
+        ).toHaveCSS("outline-color", color);
+      });
+    });
+  }
+);
+
+test.describe(
+  "Should render SimpleColorPicker component and check events",
```
Comment (ZhuoyuJin) [inline src/components/simple-color-picker/simple-color-picker.pw.tsx:408]:
```suggestion
  "check events for SimpleColorPicker component",
```

File: src/components/simple-color-picker/simple-color-picker.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,810 @@
...
+        ).toHaveCSS("max-width", cssWidth);
+      });
+    });
+
+    ([
+      ["300", "75", 89, 33, 233],
+      ["100", "60", 321, 265, 1],
+    ] as [string, string, number, number, number][]).forEach(
+      ([maxWidth, childWidth, bottomLess, topLess, leftLess]) => {
+        test(`should render with childWidth prop set to ${maxWidth}`, async ({
```
Comment (ZhuoyuJin) [inline src/components/simple-color-picker/simple-color-picker.pw.tsx:280]:
```suggestion
        test(`should render properly with childWidth prop set to ${childWidth} and maxWidth prop set to ${maxWidth} `, async ({
```

File: src/components/simple-color-picker/simple-color-picker.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,810 @@
...
+
+    await expect(focusedColor).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 255, 255) 0px 0px 0px 2px inset"
+    );
+  });
+});
+
+test.describe(
+  "Should render SimpleColorPicker component and check functionality",
```
Comment (ZhuoyuJin) [inline src/components/simple-color-picker/simple-color-picker.pw.tsx:123]:
```suggestion
  "check functionality for SimpleColorPicker component",
```
**(non-blocking suggestion)**: As the below test descriptions all have words "should render", feel a bit duplicated here.

File: playwright/components/simple-color-picker/locators.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+// component preview locators
+export const SIMPLE_COLOR_PICKER = '[data-component="simple-color-picker"]';
+
+export default { SIMPLE_COLOR_PICKER };
```
Comment (ZhuoyuJin) [inline playwright/components/simple-color-picker/locators.ts:4]:
```suggestion
export default SIMPLE_COLOR_PICKER;
```

File: playwright/components/simple-color-picker/index.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import type { Page } from "@playwright/test";
+import { SIMPLE_COLOR_PICKER } from "./locators";
+
+// component preview locators
+export const simpleColorPickerLegend = (page: Page) =>
+  page.locator(SIMPLE_COLOR_PICKER).locator("legend");
+
+export default {
+  simpleColorPickerLegend,
+};
```
Comment (ZhuoyuJin) [inline playwright/components/simple-color-picker/index.ts:10]:
```suggestion
export default simpleColorPickerLegend;
```

File: playwright/components/simple-color-picker/index.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import type { Page } from "@playwright/test";
```
Comment (ZhuoyuJin) [inline playwright/components/simple-color-picker/index.ts:1]:
```suggestion
import { Page } from "@playwright/test";
```

**Review by ZhuoyuJin — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Pull request merged**